yarn.config.cjs

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+/*!
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/** @type {import('@yarnpkg/types')} */
+const { defineConfig } = require('@yarnpkg/types');
+const fg = require('fast-glob');
+
+/**
+ * The workspace object used in the constraints function
+ * @typedef {import('@yarnpkg/types').Yarn.Constraints.Workspace} Workspace
+ */
+
+module.exports = defineConfig({
+    async constraints({ Yarn }) {
+        /**
+         * Fetch a list of all the component workspaces using a glob pattern
+         * @type {string[]} components
+         */
+        const components = fg.sync('packages/*', {
+            cwd: __dirname,
+            onlyDirectories: true,
+        });
+
+        /**
+         * This function checks the workspace for any local package references
+         * and ensure that the devDependencies are up-to-date with the latest version
+         * currently in the project
+         * @param {Workspace} workspace
+         * @returns {void}
+         */
+        function validateLocalPackages(workspace) {
+            // Return early if the workspace does not have any peerDependencies
+            if (!workspace.manifest.peerDependencies) {
+                return;
+            }
+
+            // Start by filtering out the local packages from the external dependencies
+            const localPackages = Object.keys(
+                workspace.manifest.peerDependencies
+            )
+                .filter((pkg) => Yarn.workspace({ ident: pkg }))
```

Comment:
`Yarn.workspace({ ident: pkg })` is called twice on a single package here. can we reuse the result and call it once?
```js
const localPackages = Object.keys(workspace.manifest.peerDependencies)
    .map((pkg) => Yarn.workspace({ ident: pkg }))
    .filter(Boolean);
```

Reply:
See my note on using the standard constraints syntax and tooling. This is _not_ a node script. https://yarnpkg.com/features/constraints

Reply:
`Yarn.workspace({ ident: pkg })` returns a filtered result based on the object keys specific to the workspace passed into the function. You _can't_ call it "once" as you say and get the same result.